### PR TITLE
move genericapiserver authenticator and authorizer factories

### PR DESCRIPTION
**Pull request by deads2k**
straight move to `apiserver`

### CODE REVIEW

**Comment by deads2k** [comment]
approved based on #40363

File: pkg/genericapiserver/server/config.go [hunk 1]
```diff
@@ -43,17 +43,17 @@ import (
 	"k8s.io/apimachinery/pkg/version"
 	"k8s.io/apiserver/pkg/admission"
 	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/apiserver/pkg/authentication/authenticatorfactory"
```
Comment (sttts) [inline pkg/genericapiserver/server/config.go]: Anything that speak against k8s.io/apiserver/pkg/authentication/**factory** ?

File: test/integration/framework/master_utils.go [hunk 2]
```diff
@@ -214,15 +214,15 @@ func startMasterOrDie(masterConfig *master.Config, incomingServer *httptest.Serv
 		Groups: []string{user.SystemPrivilegedGroup},
 	}
 
-	tokenAuthenticator := authenticator.NewAuthenticatorFromTokens(tokens)
+	tokenAuthenticator := authenticatorfactory.NewAuthenticatorFromTokens(tokens)
```
Comment (sttts) [inline test/integration/framework/master_utils.go]: NewFromTokens

File: pkg/genericapiserver/server/config.go (hunk 1 above)
Comment (deads2k) [inline pkg/genericapiserver/server/config.go]: > Anything that speak against k8s.io/apiserver/pkg/authentication/factory ?

You end up importing a package called `factory`.  I think it's the same reason for `net/http/httptest`

**Comment by sttts** [comment]
lgtm. Naming can be improved, see comments.

File: test/integration/framework/master_utils.go (hunk 2 above)
Comment (deads2k) [inline test/integration/framework/master_utils.go]: > NewFromTokens

Want it changed during the move or can I convince you that we start a followup issue tracking renames and package moves for cleaning up after we carve it all up?

Comment (sttts) [inline test/integration/framework/master_utils.go]: If it causes further rebase trouble, a follow-up is fine.

File: pkg/genericapiserver/server/config.go (hunk 1 above)
Comment (sttts) [inline pkg/genericapiserver/server/config.go]: would aliasing it as authenticatorfactory then ;) Anyway, a nit.

**Comment by sttts** [comment]
We need a big rename-everything-in-staging-in-a-sane-way follow-up step. People will hate us.

File: test/integration/framework/master_utils.go (hunk 2 above)
Comment (deads2k) [inline test/integration/framework/master_utils.go]: ah, only a couple call sites and they look low traffic.  updated.  Also opened https://github.com/kubernetes/kubernetes/issues/40517

**Comment by deads2k** [comment]
> We need a big rename-everything-in-staging-in-a-sane-way follow-up step. People will hate us.

Pretty sure they already do.  "why isn't this thing that didn't exist two weeks ago done and why does it have an unstable API!?"

**Comment by deads2k** [comment]
@k8s-bot bazel test this

**Pull request merged**